### PR TITLE
[fix] (training): wire gradient accumulation config

### LINKED ISSUE #366: gradient_accumulation_steps
当前 YAML 值不接入 Accelerate/DeepSpeed，Accelerator() 创建时没有从 cfg 传入；DeepSpeed 配置里固定为 1。

### PR DESCRIPTION
**Pull request by Travor278**
## Description

Wires `trainer.gradient_accumulation_steps` into Accelerate initialization for the VLA, VLA/VLM co-training, and VLM training entry points. The accelerator is now created inside `main(cfg)` after the merged trainer config is available, so YAML and CLI overrides affect `accelerator.gradient_accumulation_steps` and the total batch-size calculation.

The shared ZeRO-2 DeepSpeed config now defers gradient accumulation to Accelerate with `"auto"`. The Gemma4 and MiniCPM launch notes were updated accordingly, and MiniCPM now passes `GRAD_ACCUM=1` by default to preserve its documented effective batch size.

## Motivation

Closes #366

`trainer.gradient_accumulation_steps` was present in YAML and launch examples, but the training modules created `Accelerator(deepspeed_plugin=...)` at module import time before config parsing. That made the trainer-side setting ineffective for Accelerate, while the shared DeepSpeed config also pinned gradient accumulation to `1`.

## Changes

- Add `starVLA.training.accelerator_utils.create_accelerator(cfg)` to read and validate `trainer.gradient_accumulation_steps`.
- Move accelerator creation from module import time into `main(cfg)` for VLA, co-training, and VLM training entry points.
- Change `starVLA/config/deepseeds/ds_config.yaml` to `"gradient_accumulation_steps": "auto"`.
- Update Gemma4/MiniCPM launcher comments and preserve MiniCPM's GA=1 default explicitly.
- Add focused coverage for accelerator config wiring and keep the single-process checkpoint test Windows-safe.

## Testing

- [ ] Local training for N steps without errors
- [ ] Benchmark evaluation results (**required** for framework / benchmark changes, see table below)

| Benchmark | Metric | Result |
|-----------|--------|--------|
| N/A | N/A | Not a framework or benchmark change |

- **Checkpoint**: N/A
- **Config**: `starVLA/config/deepseeds/ds_config.yaml`
- **Reproduce**:
  - `python tests/test_training_accelerator_config.py`
  - `PYTHONPATH=. PYTHONUTF8=1 python tests/test_single_process_dist_safety.py`
  - `python -m py_compile starVLA/training/accelerator_utils.py starVLA/training/train_starvla.py starVLA/training/train_starvla_cotrain.py starVLA/training/train_starvlm.py tests/test_training_accelerator_config.py tests/test_single_process_dist_safety.py`
  - `git diff --check`
  - `docker run --rm -v "${PWD}:/workspace/starVLA" -w /workspace/starVLA starvla-train:latest bash -lc "python tests/test_training_accelerator_config.py"`
  - `docker run --rm -v "${PWD}:/workspace/starVLA" -w /workspace/starVLA starvla-train:latest bash -lc "python - <<'PY' ... PY"` verified `gradient_accumulation_steps 3` and `DistributedType.DEEPSPEED`.

## Type of Change

- [x] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] New framework / benchmark integration
- [ ] Breaking change
- [ ] Documentation only
- [ ] Refactor (no behavior change)

## Checklist

- [x] No unrelated changes mixed in
- [ ] New features have example config in `examples/`
- [x] No secrets, API keys, or large binaries committed
- [ ] Files stay isolated — no unnecessary modification to shared modules (`starVLA/dataloader/`, `starVLA/training/`, `starVLA/config/`)
- [x] No modifications to shared files, or justified above
- [ ] If adding framework/benchmark: checkpoint uploaded to personal HF (public)

## Screenshots / Logs (optional)

Unit and compile checks pass locally in the Python 3.11 environment with torch 2.7.1+cu128 and accelerate 1.5.2.

Docker smoke with the `starvla-train:latest` image also passes for the accelerator config path and confirms the real DeepSpeed-backed accelerator reads `gradient_accumulation_steps=3`. A full local GPU training step was not run because the local RTX 5070 Laptop GPU is reported as CUDA capability `sm_120`, while the current training image's PyTorch build supports up to `sm_90`.

### CODE REVIEW

**Comment by JinhuiYE** [comment]
Hi, huge thx for your contribution. I will review it this weekend.

**Comment by yeahjack** [comment]
@codex review
